addons/website_forum/models/forum.py

```diff
@@ -1044,9 +1071,9 @@ def _check_general_rights(self, vals={}):
 
     def _check_karma_rights(self, upvote=None):
         # karma check
-        if upvote and not self.post_id.can_upvote:
+        if upvote and not self.post_id.can_upvote and not self.post_id.already_downvoted:
```

Comment:
Error don't match your if now

Reply:
? You mean that it shouldn't be an AccessError anymore ? Why not, the 'if' ignore the karma rights if the user already voted down.